server/app/jobs/container_cleanup_job.rb

```diff
@@ -26,4 +26,28 @@ def destroy_deleted_containers
       c.destroy
     end
   end
+
+  def terminate_ghost_containers
+    Container.unscoped.where(:grid_service_id.ne => nil).each do |c|
+      if c.grid_service.nil?
+        terminate_ghost_container(c)
+      end
+    end
+  end
+
+  # @param [Docker::Container] container
+  def terminate_ghost_container(container)
+    if container.host_node
+      info "terminating #{container.to_path}"
+      terminator = Docker::ServiceTerminator.new(container.host_node)
+      terminator.request_terminate_service(
+        container.grid_service_id.to_s, container.instance_number, {lb: true}
```

Comment:
This GC runs once a minute.. so this disaster to happen user must first remove a service so that it will leave containers to nodes and then spin up a new service that has exactly the same name? While testing this might happen but with more serious workloads.. not so sure?